website/docs/r/monitor_data_collection_rule_association.html.markdown

```diff
@@ -0,0 +1,154 @@
+}
+
+resource "azurerm_subnet" "example" {
+  name                 = "subnet"
+  resource_group_name  = azurerm_resource_group.example.name
+  virtual_network_name = azurerm_virtual_network.example.name
+  address_prefixes     = ["10.0.2.0/24"]
+}
+
+resource "azurerm_network_interface" "example" {
+  name                = "nic"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+
+  ip_configuration {
+    name                          = "internal"
+    subnet_id                     = azurerm_subnet.example.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+resource "azurerm_linux_virtual_machine" "example" {
+  name                = "machine"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+  size                = "Standard_B1ls"
+  admin_username      = "adminuser"
+  network_interface_ids = [
+    azurerm_network_interface.example.id,
+  ]
+
+  admin_password = "example-Password@7890"
+
+  disable_password_authentication = false
+
+  os_disk {
+    caching              = "ReadWrite"
+    storage_account_type = "Standard_LRS"
+  }
+
+  source_image_reference {
+    publisher = "Canonical"
+    offer     = "UbuntuServer"
+    sku       = "16.04-LTS"
+    version   = "latest"
+  }
+}
+
+resource "azurerm_monitor_data_collection_rule" "example" {
+  name                = "example-dcr"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+  destinations {
+    azure_monitor_metrics {
+      name = "example-destination-metrics"
+    }
+  }
+  data_flow {
+    streams      = ["Microsoft-InsightsMetrics"]
+    destinations = ["example-destination-metrics"]
+  }
+}
+
+resource "azurerm_monitor_data_collection_endpoint" "test" {
+  name                = "example-dce"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+}
+
+# associate to a Data Collection Rule
+resource "azurerm_data_collection_rule_association" "example1" {
+  name                    = "example1-dcra"
+  target_resource_id      = azurerm_linux_virtual_machine.example.id
+  data_collection_rule_id = azurerm_monitor_data_collection_rule.example.id
+  description             = "example"
+}
+
+# associate to a Data Collection Endpoint
+resource "azurerm_data_collection_rule_association" "example1" {
+  name                        = "configurationAccessEndpoint"
+  target_resource_id          = azurerm_linux_virtual_machine.example.id
+  data_collection_endpoint_id = azurerm_monitor_data_collection_endpoint.example.id
+  description                 = "example"
+}
+
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name which should be used for this Data Collection Rule. Changing this forces a new Data Collection Rule to be created.
```

Comment:
if this cannot be configured to anything but configurationAccessEndpoint should we even be exposing the name? maybe we should remove it>

Reply:
they it sounds like this property should be optional and only required when data_collection_rule_id is configured, and if it is we populate it with configurationAccessEndpoint ?